dozer/cogs/tba.py

```diff
@@ -37,100 +41,268 @@ def __init__(self, bot):
-                         url='https://www.thebluealliance.com/team/{}'.format(team_num),
-                         icon_url='https://frcavatars.herokuapp.com/get_image?team={}'.format(team_num))
-            e.add_field(name='Name', value=team_data.nickname)
-            e.add_field(name='Rookie Year', value=team_data.rookie_year)
-            e.add_field(name='Location',
-                        value='{0.city}, {0.state_prov} {0.postal_code}, {0.country}'.format(team_data))
-            e.add_field(name='Website', value=team_data.website)
-            e.add_field(name='TBA Link', value='https://www.thebluealliance.com/team/{}'.format(team_num))
-            e.set_footer(text='Triggered by ' + ctx.author.display_name)
-            await ctx.send(embed=e)
-        else:
-            raise BadArgument("Couldn't find data for team {}".format(team_num))
+            team_data = await self.session.team(team_num)
+        except aiotba.http.AioTBAError:
+            raise BadArgument(f"Couldn't find data for team {team_num}.")
+        if team_data.city is None:
+            raise BadArgument("team {} exists, but has no information!".format(team_num))
+
+        try:
+            team_district_data = await self.session.team_districts(team_num)
+            if team_district_data:
+                team_district = max(team_district_data, key=lambda d: d.year)
+        except aiotba.http.AioTBAError:
+            team_district_data = None
+
+        e = discord.Embed(color=discord.Color.blurple(),
+                          title='FIRST® Robotics Competition Team {}'.format(team_num),
+                          url='https://www.thebluealliance.com/team/{}'.format(team_num))
+        e.set_thumbnail(url='https://frcavatars.herokuapp.com/get_image?team={}'.format(team_num))
+        e.add_field(name='Name', value=team_data.nickname)
+        e.add_field(name='Rookie Year', value=team_data.rookie_year)
+        e.add_field(name='Location',
+                    value='{0.city}, {0.state_prov} {0.postal_code}, {0.country}'.format(team_data))
+        e.add_field(name='Website', value=team_data.website)
+        if team_district_data:
+            e.add_field(name='District', value=f"{team_district.display_name} [{team_district.abbreviation.upper()}]")
+        e.add_field(name='Championship', value=team_data.home_championship[max(team_data.home_championship.keys())])
+        e.set_footer(text='Triggered by ' + ctx.author.display_name)
+        await ctx.send(embed=e)
 
     team.example_usage = """
     `{prefix}tba team 4131` - show information on team 4131, the Iron Patriots
     """
 
+    @tba.command()
+    @bot_has_permissions(embed_links=True)
+    async def eventsfor(self, ctx, team_num: int, year: int = None):
+        """Get the events a team is registered for a given year. Defaults to current (or upcoming) year."""
+        try:
+            events = await self.session.team_events(team_num, year=year)
```

Comment:
Passing `year=None` fetches all events from the team's history. The `year = year or ...` should happen before this call.